framework/src/main/java/org/checkerframework/framework/type/GenericAnnotatedTypeFactory.java

```diff
@@ -2219,4 +2221,94 @@ public AnnotatedTypeMirror getDefaultValueAnnotatedType(TypeMirror typeMirror) {
                 TreeUtils.getDefaultValueTree(typeMirror, processingEnv), defaultValue, false);
         return defaultValue;
     }
+
+    /**
+     * Return the string representation of contract annotations (that is, pre- and post-conditions)
+     * for the given AMethod. Does not modify the AMethod.
+     *
+     * @param m AFU representation of a method
+     * @return precondition annotations for the method
+     */
+    public List<String> getContractAnnotations(AMethod m) {
+        List<String> preconds = getPreconditionAnnotations(m);
+        List<String> postconds = getPostconditionAnnotations(m, preconds);
+        List<String> result = preconds;
+        result.addAll(postconds);
+        return result;
+    }
+
+    /**
+     * Return the string representation of precondition annotations for the given AMethod. Does not
+     * modify the AMethod.
+     *
+     * @param m AFU representation of a method
+     * @return precondition annotations for the method
+     */
+    public List<String> getPreconditionAnnotations(AMethod m) {
+        List<String> result = new ArrayList<>();
+        for (Map.Entry<VariableElement, AField> entry : m.getPreconditions().entrySet()) {
+            VariableElement elt = entry.getKey();
+            AField afield = entry.getValue();
+
+            String precondAnno = getPreconditionAnnotation(elt, afield);
+            if (precondAnno != null) {
+                result.add(precondAnno);
+            }
+        }
+        return result;
+    }
+
+    // TODO: Implement this here in this class, reading @PreconditionAnnotation meta-annotations.
+    // That would be more general and would eliminate most but not all of the code in overriding
+    // implementations.
+    /**
+     * Return the string representation of a precondition annotation for the given field.
```

Comment:
It's not clear how to convert from field to a precondition annotation.  Can you expand this comment?